frontend/src/components/Signup.js

```diff
@@ -0,0 +1,156 @@
+import React, { useState, useEffect } from 'react'
+import { useSelector, useDispatch, batch } from 'react-redux'
+import { useHistory } from 'react-router-dom'
+import styled from 'styled-components/macro'
+import Swal from 'sweetalert2'
+
+import user from '../reducers/user'
+
+import { API_URL } from '../reusable/urls'
+
+const Signup = () => {
+  const [username, setUsername] = useState('')
+  const [password, setPassword] = useState('')
+  const [mode, setMode] = useState(null)
+
+  const accessToken = useSelector(store => store.user.accessToken)
+  const dispatch = useDispatch()
+  const history = useHistory()
+  
+  useEffect(() => {
+    if (accessToken) {
+        history.push('/')
+    }
+}, [accessToken, history])
+
+  const onFormSubmit = (e) => {
+    e.preventDefault()
+
+    const options = {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json'
+      },
+      body: JSON.stringify({
+        username,
+        password
+      })
+    }
+    fetch(API_URL(mode), options)
+      .then(res => res.json())
+      .then(data => {
+        if (data.success) {
+          batch(() => {
+            dispatch(user.actions.setUsername(data.username))
+            dispatch(user.actions.setAccessToken(data.accessToken))
+            dispatch(user.actions.setErrors(null))
+          })
+        } else {
+          if(data.error.code === 11000) {
+            Swal.fire('Please choose another username.')
+            dispatch(user.actions.setErrors(data))
+          } else {
+            Swal.fire(`Username: ${data.error.errors.username.message}`)
+            dispatch(user.actions.setErrors(data))
+          }
+        }
+      })
+      .catch()
+    }
+
+  return (
+    <Form
+      onSubmit={onFormSubmit}
+    >
+      <Title>
+        Sign up!
+      </Title>
+      <Label
+        htmlFor='username'
+      >
+        Username
+      </Label>
+      <Input
+        id='username'
+        type='text'
+        required
+        minlength='5'
+        maxlength='18'
+        value={username}
+        onChange={(e) => setUsername(e.target.value)}
+      />
+      <Label
+        htmlFor='password'
+      >
+        Password
+      </Label>
+      <Input
+        id='password'
+        type='password'
+        required
+        value={password}
+        onChange={(e) => setPassword(e.target.value)}
+      />
+      <Button
+        type='submit'
+        onClick={() => setMode('signup')}
+      >
+        SIGN UP
+      </Button>
+    </Form>
+  )
+}
```

Comment:
Signup is very similar to login, could maybe be some component with props instead of two different pages?